lms/djangoapps/courseware/models.py

```diff
@@ -248,3 +251,56 @@ class Meta(object):   # pylint: disable=missing-docstring
 
     field = models.CharField(max_length=255)
     value = models.TextField(default='null')
+
+
+SCORE_CHANGED = Signal(
+    providing_args=[
+        'points_possible',  # Maximum score available for the exercise
+        'points_earned',   # Score obtained by the user
+        'user',  # User object representing the student
+        'course',  # CourseLocator object representing the course
+        'usage'  # BlockUsageLocator indicating the courseware instance
+    ]
+)
+
+
+@receiver(score_set)
+def submissions_score_set_handler(sender, **kwargs):  # pylint: disable=unused-argument
```

Comment:
My opinion on kwargs is exactly the same as yours (although possibly with less diplomatic language - I'm a Java developer at heart). Unfortunately, this is the way that the Django signals mechanism is designed. From the signals documentation (https://docs.djangoproject.com/en/1.8/topics/signals):

"Notice that the function takes a sender argument, along with wildcard keyword arguments (**kwargs); all signal handlers must take these arguments.
We’ll look at senders a bit later, but right now look at the **kwargs argument. All signals send keyword arguments, and may change those keyword arguments at any time."

Personally I think that this is a pretty terrible interface design (your signal handler must be robust enough to accept any possible argument that may change at any time, yet must still somehow be able to do something useful), but it looks like it's the cost of doing business with signals...